src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -110,11 +114,22 @@ namespace cryptonote
 
     m_block_download_max_size = command_line::get_arg(vm, cryptonote::arg_block_download_max_size);
     m_sync_pruned_blocks = command_line::get_arg(vm, cryptonote::arg_sync_pruned_blocks);
+    m_span_time = command_line::get_arg(vm, cryptonote::arg_span_limit);
 
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  void t_cryptonote_protocol_handler<t_core>::calculate_dynamic_span(double blocks_per_seconds)
+  {
+    MINFO("m_bss : " << m_bss << ", blocks_per_seconds : " << blocks_per_seconds << ", m_span_limit : " << m_span_limit);
+    m_span_limit = (m_bss && blocks_per_seconds) ? (( blocks_per_seconds * 60 * m_span_time ) / m_bss) : BLOCK_QUEUE_NSPANS_THRESHOLD;
+    if (m_span_limit < BLOCK_QUEUE_NSPANS_MINIMUM)
+      m_span_limit = BLOCK_QUEUE_NSPANS_MINIMUM;
+    MINFO("calculated dynamic span limit is span_limit : " << m_span_limit);
```

Comment:
its default "what can be synced within 2 minutes".

do you mean, a cap on if a user decides to set it to, say, 60 minutes?

Reply:
I don't see how `m_span_limit` is what can be synced within 2 minutes - the time portion tracks just the DB sync time. And it's divided by this `bss` value.

Reply:
I guess the idea is fill the DB time as much as possible, which should work, maybe I'm being too harsh here.

Reply:
Bss = block sync size (how many blocks in  span)

The way its intended to work:

db sync time returns how many blocks synced per second
using this calculation, we take the block sync size (lets say 20blocks per span) do math to calculate how many spans to queue to stay 2mins ( default span-limit) ahead in queue

Reply:
The timings track just the DB sync time, and it could result in rather large memory usage. The big counter-argument is that the user can put `0` here and use the current value.